library-best-practices/JLBP-7.md

```diff
@@ -0,0 +1,35 @@
+[JLBP-7] Make breaking transitions easy
+---------------------------------------
+
+When a library makes an in-place breaking change (meaning that the Java package
+is not renamed - see [JLBP-6](JLBP-6.md) for more discussion on renaming rules),
+this can create a lot of work for consumers to adapt to the change. If a change
+is "atomic" (meaning that the old version of something like a method is removed
+at the same time the new version is added), all usages of the old method must be
+converted to the new method in a single change to each consumer's codebase. If
+the usage is wide enough and especially if it spans multiple packages, the cost
```

Comment:
Is "multiple packages" the most relevant metric? I'd expect the main complications to be things like multiple owners or multiple release cycles.

Reply:
Yeah it's not necessarily simply multiple artifacts, but that it is artifacts across multiple owners and multiple levels of the tree. I will rework this.